src/backend/distributed/sql/citus--8.0-1.sql

```diff
@@ -954,11 +954,11 @@ CREATE FUNCTION worker_apply_sequence_command(text)
 COMMENT ON FUNCTION worker_apply_sequence_command(text)
     IS 'create a sequence which products globally unique values';
 
-CREATE FUNCTION isolate_tenant_to_new_shard(table_name regclass, tenant_id "any", cascade_option text DEFAULT '')
+CREATE FUNCTION isolate_tenant_to_new_shard(table_name regclass, tenant_id "any", cascade_option text DEFAULT '', is_blocking boolean DEFAULT true)
```

Comment:
would be better to stick to the shard_transfer_mode pattern followed by citus_move_shard_placement, citus_copy_shard_placement, rebalance_table_shards, and citus_split_shard_by_split_points.

It's not ideal, but at least consistent. We'll likely remove it once non-blocking no longer causes update/delete errors when there are no primary keys.

Reply:
btw, we should never change past SQL files since they are not revisited when upgrading. Instead we need to redefine isolate_tenant_to_new_shard in an upgrade script, which will also be executed for new installations.